### PR TITLE
Turbo replication changes

### PR DESCRIPTION
**Pull request by anshuldavid13**
PR for https://github.com/GoogleCloudPlatform/dotnet-docs-samples/issues/1509

### CODE REVIEW

File: storage/api/Storage.Samples/SetRPODefault.cs
Comment (amanda-tarafa) [inline storage/api/Storage.Samples/SetRPODefault.cs:21]:
```suggestion
public class SetRpoDefaultSample
```
Here and everywhere, including file names.

File: storage/api/Storage.Samples/SetRPODefault.cs
Comment (amanda-tarafa) [inline storage/api/Storage.Samples/SetRPODefault.cs:24]: When it's just a few parameters, there's no need to put them on the next line.
```suggestion
    public void SetRPODefault(string bucketName = "your-unique-bucket-name")
```

File: storage/api/Storage.Samples/SetRPODefault.cs
Comment (amanda-tarafa) [inline storage/api/Storage.Samples/SetRPODefault.cs:31]: Does setting it to default means that it is disabled? Just double checking this.

File: storage/api/Storage.Samples.Tests/SetRPODefaultTest.cs
Comment (amanda-tarafa) [inline storage/api/Storage.Samples.Tests/SetRPODefaultTest.cs:34]: Should you create a new bucket for this test? I don't remember what the other tests that change the metadata do, but maybe we shouldn't change the metadata for the generic bucket as it might make it useless for tests that come later.

File: storage/api/Storage.Samples/SetRPOAsyncTurbo.cs
Comment (amanda-tarafa) [inline storage/api/Storage.Samples/SetRPOAsyncTurbo.cs:20]:
```suggestion
```

File: storage/api/Storage.Samples.Tests/SetRPODefaultTest.cs
Comment (amanda-tarafa) [inline storage/api/Storage.Samples.Tests/SetRPODefaultTest.cs:32]: Maybe use the GetBucketRpo sample instead?

File: storage/api/Storage.Samples/CreateBucketWithTurboReplication.cs
Comment (amanda-tarafa) [inline storage/api/Storage.Samples/CreateBucketWithTurboReplication.cs:29]: I think you can use the CreateBucket overload that takes a Bucket and set all fields from the start, that way you don't have to make to calls (create and update) but just the create one.

File: storage/api/Storage.Samples.Tests/CreateBucketWithTurboReplicationTest.cs
Comment (amanda-tarafa) [inline storage/api/Storage.Samples.Tests/CreateBucketWithTurboReplicationTest.cs:35]: Add a method on the fixture, something like GenerateBucketName, and add that bucket name into the TempBuckets (see the method CreateBucket in the fixture), so this bucket will be removed on clean up.

**Review by amanda-tarafa — CHANGES_REQUESTED** [review]
Fine overall, a few request changes.

And some comments will apply everywhere, not just where I made it.

File: storage/api/Storage.Samples/SetRPODefault.cs
Comment (anshuldavid13) [reply storage/api/Storage.Samples/SetRPODefault.cs:29]: yes 
image

File: storage/api/Storage.Samples.Tests/SetRPODefaultTest.cs
Comment (anshuldavid13) [reply storage/api/Storage.Samples.Tests/SetRPODefaultTest.cs]: I could see these tests creating fresh temporary buckets in the instance for each test run. Am I missing something?

**Comment by ddelgrosso1** [comment]
@anshuldavid13 once this is approved please go ahead and merge it when you can. We are trying to get this feature / samples released this week. Thanks!

File: storage/api/Storage.Samples/CreateBucketWithTurboReplication.cs
Comment (amanda-tarafa) [inline storage/api/Storage.Samples/CreateBucketWithTurboReplication.cs:30]: we need to add this bucket.Name to the temp buckets for cleanup

```_fixture.TempBucketNames.Add(bucket.Name)```

File: storage/api/Storage.Samples/SetRpoAsyncTurbo.cs
Comment (amanda-tarafa) [inline storage/api/Storage.Samples/SetRpoAsyncTurbo.cs:24]:
```suggestion
        // Enabling turbo replication requires a bucket with dual-region configuration.
        // You can read more about dual-region configuration here: https://cloud.google.com/storage/docs/locations#location-dr
```

File: storage/api/Storage.Samples.Tests/SetRpoAsyncTurboTest.cs
Comment (amanda-tarafa) [inline storage/api/Storage.Samples.Tests/SetRpoAsyncTurboTest.cs:33]:
```suggestion
        // Enabling turbo replication requires a bucket with dual-region configuration.
```

Does the generic bucket has dual-region configuration? Else, this test will always fail.

And also, this test and the one setting Default RPO may run at the same time, they'd both be changing the RPO setting for the same bucket so the last one that changes the value will "win" and the other test will fail.

In both these tests you need to create a new bucket and use that one instead of the generic one. Also, for this test in particular you need to make sure that the bucket you have created is dual-region.

File: storage/api/Storage.Samples/SetRpoAsyncTurbo.cs
Comment (amanda-tarafa) [reply storage/api/Storage.Samples/SetRpoAsyncTurbo.cs:24]: And change to this everywhere else.

**Review by amanda-tarafa — CHANGES_REQUESTED** [review]
One nit, and one change we need to avoid tests failing.

**Review by amanda-tarafa — APPROVED** [review]
LGTM

I'll run tests and I can merge once it's green.

**Comment by amanda-tarafa** [comment]
Error is an unrelated transient Monitoring error. Merging

**Pull request merged**
